nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/pgp/PGPControllerService.java

```diff
@@ -0,0 +1,893 @@
+                if (!(head instanceof PGPEncryptedDataList)) {
+                    throw new IOException("Input stream does not contain PGP encrypted data.");
+                }
+            }
+
+            try {
+                final Iterator objects = ((PGPEncryptedDataList) head).getEncryptedDataObjects();
+                PGPEncryptedData data;
+
+                while (objects.hasNext()) {
+                    Object obj = objects.next();
+                    InputStream clearInput;
+
+                    if (obj instanceof PGPPublicKeyEncryptedData) {
+                        data = (PGPEncryptedData) obj;
+                    } else if (obj instanceof PGPPBEEncryptedData) {
+                        data = (PGPEncryptedData) obj;
+                    } else {
+                        throw new IOException("Expected encrypted data");
+                    }
+                    clearInput = options.getInputStream(data);
+
+                    JcaPGPObjectFactory plainFactory = new JcaPGPObjectFactory(clearInput);
+                    Object plain = plainFactory.nextObject();
+
+                    if (plain instanceof PGPCompressedData) {
+                        PGPCompressedData compressed = (PGPCompressedData) plain;
+                        plain = (new JcaPGPObjectFactory(compressed.getDataStream())).nextObject();
+                    }
+
+                    if (plain instanceof PGPLiteralData) {
+                        PGPLiteralData literalObject = (PGPLiteralData) plain;
+                        try (InputStream literalInput = literalObject.getInputStream()) {
+                            Streams.pipeAll(literalInput, output);
+                        }
+                    } else {
+                        throw new PGPException("message is not a simple encrypted file - type unknown.");
+                    }
+
+                    if (data.isIntegrityProtected()) {
+                        if (!data.verify()) {
+                            throw new PGPException("Failed message integrity check");
+                        }
+                    } else {
+                        throw new PGPException("Encrypted data packet does not have integrity check.");
+                    }
+                }
+            } catch (final Exception e) {
+                throw new IOException(e.getMessage());
+            }
+        }
+    }
+
+
+    /**
+     * Read from an {@link InputStream} to generate a signature written to an {@link OutputStream}.
+     *
+     * @param input clear or cipher data
+     * @param signature to receive signature
+     * @param options used to configure sign operation
+     */
+    public void sign(InputStream input, OutputStream signature, SignOptions options) throws IOException, PGPException {
+        if (options == null || options.getPrivateKey() == null) {
+            throw new IOException("Sign operation invalid without Private Key");
+        }
+        final int KEY_ALGORITHM = 1; // wha
```

Comment:
I am guessing this means something but the comment is unclear.

Reply:
Value corrected and extraneous comment removed.